src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/matchtoterm/TermFieldRewriter.java

```diff
@@ -147,19 +149,27 @@ public boolean visit(SQLIdentifierExpr expr) {
 
     public void collect(SQLTableSource tableSource, Map<String, String> indexToType,  Map<String, String> aliases) {
         if (tableSource instanceof SQLExprTableSource) {
+
             String table = null;
-            if (((SQLExprTableSource) tableSource).getExpr() instanceof SQLIdentifierExpr) {
-                table = ((SQLIdentifierExpr) ((SQLExprTableSource) tableSource).getExpr()).getName();
-                indexToType.put(
-                    table,
-                    null
-                );
-            } else if (((SQLExprTableSource) tableSource).getExpr() instanceof SQLBinaryOpExpr) {
-                table = ((SQLIdentifierExpr)((SQLBinaryOpExpr) ((SQLExprTableSource) tableSource).getExpr()).getLeft()).getName();
-                indexToType.put(
-                    table,
-                    ((SQLIdentifierExpr)((SQLBinaryOpExpr) ((SQLExprTableSource) tableSource).getExpr()).getRight()).getName()
-                );
+            SQLExprTableSource sqlExprTableSource = (SQLExprTableSource) tableSource;
+
+            if (sqlExprTableSource.getExpr() instanceof SQLIdentifierExpr) {
+                SQLIdentifierExpr sqlIdentifier = (SQLIdentifierExpr) sqlExprTableSource.getExpr();
+                table = sqlIdentifier.getName();
+                indexToType.put(table, null);
+            } else if (sqlExprTableSource.getExpr() instanceof SQLBinaryOpExpr) {
+                SQLBinaryOpExpr sqlBinaryOpExpr = (SQLBinaryOpExpr) sqlExprTableSource.getExpr();
+                table = ((SQLIdentifierExpr) sqlBinaryOpExpr.getLeft()).getName();
+                SQLExpr rightSideOfExpression = sqlBinaryOpExpr.getRight();
+
+                // This assumes that right side of the expression is different name in query
+                if (rightSideOfExpression instanceof SQLIdentifierExpr) {
+                    SQLIdentifierExpr right = (SQLIdentifierExpr) rightSideOfExpression;
+                    indexToType.put(table, right.getName());
+                } else {
+                    throw new ParserException("Right side of the expression  " + rightSideOfExpression.toString() +
```

Comment:
add some delimiters to distinguish between the exception message and the actual expression value, e.g. `"... expression [" + value + "] is expected ..."`. Also I am a fan of using the [rfc2119](https://tools.ietf.org/html/rfc2119) terms, so in this case **must**.

Reply:
We are not sure if there are no other cases, so as of now I want to avoid "must"

Reply:
ok, makes sense.